labs/a8.md

```diff
@@ -6,274 +6,113 @@ layout: lab
+use the command `puppet parser validate hacking.pp`.
 
-Due to security issues, the Minecraft server running is only listening to
-`127.0.0.1`, which means by default you can only access a Minecraft client
-running on the VM. You have two ways to get around this. One is changing
-`server-ip` in server.properties to `0.0.0.0`, which will allow access by
-anybody. If you do this you probably want to set up a whitelist. The other
-option is SSH port forwarding. The command, if run on your machine, captures
-traffic at port 25565 on your local computer, and forwards them to port 25565 on
-your VM. The command will run forever, just leave it in the background while you
-try to connect.
+Once you have completed your manifest, you can apply the changes to your system
+with this command (assuming you cloned `decal-labs` into $HOME):
 
 ```
-ssh -NL 25565:localhost:25565 <your VM>
+sudo env "PATH=$PATH" puppet apply --modulepath=$HOME/decal-labs/a8/modules $HOME/decal-labs/a8/manifests/default.pp
 ```
 
-If you chose the first option, you can connect by typing your domain name or IP
-into minecraft, if you chose the second, you can connect to `localhost`.
-
-## Part 3: Cleanup
+Basically, we’re running puppet locally on the module we just created. 
+`default.pp` is just a file that includes the hacked module you’re editing, 
+and we need to be sure to include the path to the modules so puppet can find them.
 
-There is currently a cronjob copying the minecraft world every minute. That
-might run you out of disk space. To disable that, you should run `sudo crontab
--e` and remove the line for the backups.
+The weird little `env` part is due to a Debian security feature known as `secure_path`,
+which resets your `PATH` when you run sudo. Since `puppet` is not in this list of 
+default paths, running it with regular `sudo` won't work.
 
-Trying to run a Minecraft Server constantly might also eat up some system
-resources. You can stop and disable the minecraft systmemd unit manually if it
-is causing issues.
+Once this is done, you should be able to run the command `hack_everything` 
+(as the script is now in your path). Try it out!
 
-If you added the swapfile, you might want to remove that.
+Hints (if you need them):
 
-Another way you can clean up is with puppet. By default, puppet doesn't remove
-files it doesn't know about. However, you can use `ensure => absent` to make
-sure files are gone, and similarly for the other resource types.
+- The skeleton code has links to relevant docs
+- [Here](https://github.com/ocf/puppet/blob/4e37cbedd228d89bc2f32234dbb4fc54114faa9d/modules/ocf_desktop/manifests/suspend.pp) is an example puppet manifest that might be helpful
 
 ## Part 4: Submission
 
 Congratulations on finishing the lab!
 
-To submit, copy paste the code you have for each section into the gradescope
-submission.
+To submit, copy paste the code for hacking.pp into the Gradescope submission.
```

Comment:
It may be more convenient for everyone to turn submission into a file upload (nicer code formatting, no need to copypaste, etc)